iocore/net/quic/QUICAckFrameCreator.cc

```diff
@@ -187,11 +187,12 @@ QUICAckFrameManager::QUICAckFrameCreator::push_back(QUICPacketNumber packet_numb
     this->_should_send = true;
   }
 
-  // every 2 full-packet should send a ack frame like tcp
-  this->_size_unsend += size;
-  // FIXME: this size should be fixed with PMTU
-  if (this->_size_unsend > 2 * 1480) {
-    this->_size_unsend = 0;
+  if (!ack_only) {
+    this->_ack_eliciting_count++;
+  }
+
+  // every 2 ack-eliciting packet should send a ack frame
+  if (this->_ack_eliciting_count % 2 == 0) {
```

Comment:
Is there a case `_should_send` is cleared without sending ACK? I'm wondering we could move this block into the `if` block above.

Reply:
>Is there a case _should_send is cleared without sending ACK

No I think, See https://github.com/apache/trafficserver/blob/83a443bb2b1f3cca701c6f6087b184ee81d65078/iocore/net/quic/QUICAckFrameCreator.cc#L350-L360

Reply:
In QVC::common_send, it will check every generator will_generate_frame. For QUICAckFrameManager, it will check every creator through this api (is_ack_frame_ready). `_available ` means whether we have not ack only packet in list. It is always true when we receive ack-eliciting packet (it is under your comment line).